hedyorg/hedy
# [RU] Resolve buttons are now functional

[To be approved by Radboud NML members, for #4166 ]


The resolve function on RADboard are now functional. 
When clicked they show a prompt asking to confirm, if yes is pressed, that item is removed from the feed. 

Technical Choices:
Behind the scenes the list items are loaded dynamically by reading from a small json database. 
This is the file `radboard_error_data.json`. For now this file should serve the purpose of dummy data and testing purposes, eventually the error data should probably be stored in the dynamo database as is used on deploy. 
When yes is pressed on the resolve prompt, a js function is triggered that sends a delete request to a route which is found in the statistics.py file [remove_common_error_item()]. In here the active status of the specific error item is set to 0, this means it won't be rendered anymore in the frontend. Then the database is reloaded into memory, otherwise the frontend loads in the outdated version. 
In general, I did a bunch of reformatting and structuring code to make it nicer to work with. Most notably I split up the live stats and normal stats into two classes/modules. 

How to test:
Run hedy locally on this branch, navigate to live dashboard, click resolve button in the common error section. A prompt should show.
If no is pressed, nothing should happen. If yes is pressed that item should dissappear. 
Since the database file is changed, make sure to discard any changes made to your `radboard_error_data.json` file.
